File: Recursion/subsequence_pattern/GenerateBinaryStringWithoutConsecutiveZeros.cpp
#include <string>
#include <vector>

class GenerateBinaryStringWithoutConsecutiveZeros {
    static void helper(std::vector<std::string>& result, const std::string& current, int n) {
        if (static_cast<int>(current.size()) == n) { result.push_back(current); return; }
        helper(result, current + '1', n);
        if (current.empty() || current.back() == '1') helper(result, current + '0', n);
    }
public:
    static std::vector<std::string> validStrings(int n) {
        std::vector<std::string> result;
        helper(result, "", n);
        return result;
    }
};

File: Recursion/subsequence_pattern/GenerateBinaryStringWithoutConsecutiveZeros.py
def _helper(result: list[str], current: str, length: int) -> None:
    if len(current) == length:
        result.append(current)
        return
    _helper(result, current + "1", length)
    if not current or current[-1] == "1":
        _helper(result, current + "0", length)


def valid_strings(length: int) -> list[str]:
    result = []
    _helper(result, "", length)
    return result
